### PR TITLE
[LDS-26] Create Latitude Breadcrumb Component

### PR DESCRIPTION
**Pull request by Cytancy**
## Description:
Adds a Latitude breadcrumb component. Includes a Storybook story, tests, and documentation. Partial migration of this PR: https://github.flexport.io/flexport/flexport/pull/53242

## Jira:
https://flexport.atlassian.net/browse/LDS-26

## Manual Testing Plan:
1. Run Storybook.
2. See Breadcrumb story, observe correct behavior when clicking links. You can reset the breadcrumbs to their original state using the "Reset Breadcrumbs" button in the 'Knobs' section. See screenshot section below for example.
3. Run Jest tests for the Breadcrumb component using the following: `yarn test __tests__/Breadcrumbs-test.js`
4 Confirm all tests pass successfully.

## Screenshots:
Breadcrumbs-storybook

### CODE REVIEW

File: Breadcrumbs.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import Icon from "./Icon";
+
+type BreadcrumbProps = {|
+  /** URL a Breadcrumb item points to */
+  +href?: string,
+  /** called when the Breadcrumb item is clicked */
+  +onClick?: ?() => void,
+  /** element, text, or  number rendered within the breadcrumb */
+  // This `any` is fine, alternative to React.Node which uses React.Element<any>, disallowing boolean, null, and void cases
+  +content: string | React.Element<any> | number, // flowlint-line unclear-type:off
```
Comment (notandrewkaye) [inline Breadcrumbs.jsx]: did we decide that we can dial this back to just string?

Comment (Cytancy) [reply Breadcrumbs.jsx]: I I thought we decided to do that in monorepo via a wrapper (e.g. `longitude/Breadcrumb`) to support the DG badge. Here's the ticket: https://flexport.atlassian.net/browse/LDS-398?atlOrigin=eyJpIjoiMTNlM2ZjMDUwNWE4NDA4MzgzYTE0OWZhNDZlYzhkZWMiLCJwIjoiaiJ9

File: Breadcrumbs.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import Link from "./Link";
+import Icon from "./Icon";
+
+type BreadcrumbProps = {|
+  /** URL a Breadcrumb item points to */
+  +href?: string,
+  /** called when the Breadcrumb item is clicked */
+  +onClick?: ?() => void,
+  /** element, text, or  number rendered within the breadcrumb */
+  // This `any` is fine, alternative to React.Node which uses React.Element<any>, disallowing boolean, null, and void cases
```
Comment (notandrewkaye) [inline Breadcrumbs.jsx]: Why is this a `//` comment?

Comment (Cytancy) [reply Breadcrumbs.jsx]: I fix.

**Pull request merged**